tests/Fixer/ReturnNotation/SimplifiedNullReturnFixerTest.php

```diff
@@ -55,4 +55,44 @@ public function provideFixCases()
             ['<?php return;', "<?php return\n(\nnull\n)\n;"],
         ];
     }
+
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider provideNullableReturnTypeCases
+     * @requires PHP 7.1
+     */
+    public function test71ReturnTypes($expected, $input = null)
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public function provideNullableReturnTypeCases()
+    {
+        return [
+            ['<?php function foo(): ? /* C */ int { return null; }'],
+            ['<?php function foo(): ?int { if (false) { return null; } }'],
+            [
+                '<?php function foo(): ?int { return null; } return;',
+                '<?php function foo(): ?int { return null; } return null;',
+            ],
+            [
+                '<?php function foo() { return; } function bar(): ?A\B\C\D { return null; } function baz() { return; }',
+                '<?php function foo() { return null; } function bar(): ?A\B\C\D { return null; } function baz() { return null; }',
+            ],
+            [
+                '<?php function foo(): ?int { $bar = function() { return; }; return null; }',
+                '<?php function foo(): ?int { $bar = function() { return null; }; return null; }',
+            ],
+            [
+                '<?php function foo(): int { return; }',
```

Comment:
> Fatal error: A function with return type must return a value

Reply:
```
function foo(): int { return null; }
// Fatal error: Uncaught TypeError: Return value of foo() must be of the type integer, null returned

function foo(): int { return; }
// Fatal error: A function with return type must return a value
```
Both before and after the fix there is a Fatal error.

The reason why I wrote this test is to prove **this** PR doesn't change the (erroneous) behaviour of the fixer.

Reply:
I agree that initial code was wrong as well. IMO we shall not change one wrong code into another, but simply don't touch it.

Reply:
Done.